chacha20/Cargo.toml

```diff
@@ -32,6 +32,7 @@ hex-literal = "0.3.3"
 [features]
 std = ["cipher/std"]
 zeroize = ["cipher/zeroize"]
+neon = []
```

Comment:
It might be worth considering using a `cfg!` attribute here /cc @newpavlov 

I think the only reason to have a gated implementation at all is MSRV compatibility. We can bump to MSRV 1.61 in the next release (bumping to 1.60 has some nice additions like weak feature activation).

Semi-related issue: https://github.com/RustCrypto/sponges/pull/24#discussion_r1020815841

FWIW, we use `cfg!` attributes to gate `aes` and `pmull` intrinsics on ARMv8